pkg/bmc/access.go

```diff
@@ -46,6 +46,9 @@ type AccessDetails interface {
 
 	// Boot interface to set
 	BootInterface() string
+
```

Comment:
This is fine for a single RAID. But in many cases you have 2 RAIDs.
One for 2 or small # of disks for Boot/Operating system.
Another one, including NORAID for remaining disks.

Reply:
This applies to the whole patch.
It will be too hard to retrofit later.

Reply:
Do you have a specific proposal for a different API? As far as I can tell, the gopher cloud client for Ironic only accepts a single string. Are you proposing that it should take multiple values? Or are you proposing that the user needs to somehow specify the value?

Reply:
Doug, 
I am thinking that a variable length array will be sufficient.
First, parameter the size of array. Default 1.
The second is array of RAIDinterfaces. Each raid interface will be a tuple. First, part of tuple will be RAID type, second is list (or array) of disks in that RAID.

I recall that Ironic has a way to specify more than one RAID per node. Would be good to check how they specify it. Maybe Digmabar can comment on it.

Reply:
This is an internal API, for statically defined types that let us pass different values to ironic based on the BMC type the user gives us. I think the value being returned here is telling ironic which RAID driver to use within itself (note the values returned by the methods below), rather than specifying individual controllers on a host. Are you saying we may need to tell ironic multiple values in that case?

Reply:
Yes, this is just telling the Ironic driver which RAIDInterface to use to configure RAID on this particular machine. So e.g. for the [iDRAC driver ](https://docs.openstack.org/ironic/latest/admin/drivers/idrac.html#enabling) the options are `idrac-wsman`/`idrac`, or `no-raid` to disable RAID support altogether (or `agent`, to configure it in software). Perhaps one day there'll be an `idrac-redfish` too. But this has nothing to do with what RAID arrays are to be created, that's in #292. All arrays are configured, or not configured, through the same interface.

Reply:
Yes, I agree with Arkady. IRonic support multiple Raid configuration for RAID. We should allow use to create multiple RAID which would be helpful for configuring root disk (say RAID1) and remaining disks (RAID 0 or non-raid) depending on the RAID controller. This will certainly good feature for us as Ironic is already supporting it.
@zaneb  Yes. You understanding is correct. Above implementation is irrespective of which driver you are using whether it's idrac-wsman/idrac/idrac-redfish.

Reply:
That's exactly what's proposed in the current PR: https://github.com/metal3-io/baremetal-operator/pull/292/files#diff-2336c3885f739a6a5c66bc71e72497b3R133-R140